lib/auth/v4/awsURIencode.ts

```diff
@@ -44,8 +44,9 @@ export default function awsURIencode(
     if (typeof input !== 'string') {
         return encoded;
     }
-    for (let i = 0; i < input.length; i++) {
-        const ch = input.charAt(i);
+    const charArray = Array.from(input);
```

Comment:
( unrelated to this line, just using it for inline comments )
Idk how big the inputs can be but this func is actually quite inefficient using string concats and even worse the concat method itself ( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/concat#performance ). I could rewrite it more efficiently if this is something we want to tackle. If the inputs are only ever like 15 characters long its whatever but I believe we use this quite often and the inputs could be quite long?

Reply:
I just double checked and yes, the function is used to encode policy documents, canonical requests (they can be very long) and other but smaller resource names.

Reply:
If you know how to make it more efficiently we surely can make it, this is always a benefits to be more efficient..

Reply:
Ok let me give it a shot ( closing conversation to open another one. )